### ISSUE TITLE
Clearing/Updating Series points causes null ref if smoothing turned on

### ISSUE THREAD

**Issue by RyanBuzzInteractive** [oxyplot/oxyplot#422]
I am working on Xamarin iOS (MonoTouch). 

I am trying to clear all of the points from a series and add a fresh collection. If I clear the current list of points, I get a null ref afterwards when I click on the plot (at Oxyplot's LineSeries 'GetNearestPoint' method). 

If I don't clear the list and just append the points, the crash is avoided but strangely they show up on a separate line, almost like a new series.

I even tried clearing the PlotModel's list of series and adding a fresh one with the new points, but find the same null ref crash.

It seems that the PlotView is trying to find the 'nearest point' from the old points which no longer exist, and my new points are acting like a separate collection. I can't see why this would be happening.

I have tracked it down to the 'smooth' option on the series. Switching smoothing OFF eliminates the problem. The line that triggers the null ref in LineSeries.GetNearestPoint()  is in this block, when it tries to access the result's item:

``` csharp
if (interpolate && this.Smooth && this.SmoothedPoints != null)
{
var result = this.GetNearestInterpolatedPointInternal(this.SmoothedPoints, point);
result.Text = this.Format( this.TrackerFormatString, 
result.Item, // Crashes here!!
this.Title, 
this.XAxis.Title ?? XYAxisSeries.DefaultXAxisTitle,
this.XAxis.GetValue(result.DataPoint.X),
this.YAxis.Title ?? XYAxisSeries.DefaultYAxisTitle
this.YAxis.GetValue(result.DataPoint.Y));
}
```

At a guess, when the 'normal' point collection is modified, the smoothed points aren't updated?? I am using the Xamarin component so I had to just make my 'own' class which is a duplicate of LineSeries and step through it, but that means I cant dive any deeper very easily and I don't have time right now.

Commenting out that whole block seems to make the problem go away. I might be being thick, but it seems like 'result' isn't actually being used for anything?! 

Cheers,
Ryan

**Comment by tibel** [comment]
Please try to add a `lock (plotModel.SyncRoot)` around the code that manipulates the plot model.
You should always use a lock when manipulating the plotModel.

**Comment by RyanBuzzInteractive** [comment]
Cool, thanks. I'll give that a go and report back. Cheers dude :)

**Comment by RyanBuzzInteractive** [comment]
Hi @tibel , i'm afraid that hasn't made a difference. Thanks for pointing it out, I hadn't know about the need to lock the model when adjusting the collection, but I think in this case it is something else. 

Given that it doesn't happen when I turn smoothing off, I am inclined to think it's something to do with the sync between the 'original' point collection and the derived smoothed point collection. 

I don't think it's a race condition as it only happens on the next click after changing the points. Some of the heavier calculation logic is done in an async Task, but I have put in place the locks you suggested around anything which touches the Model or its points etc.

**Comment by RyanBuzzInteractive** [comment]
I can confirm that the collection (this.SmoothedPoints) is empty after clearing the 'normal' points list and adding new points.

**Comment by RyanBuzzInteractive** [comment]
I tried hacking in a 'Reset Smoothed Points' call inside the ' if (interpolate && this.Smooth && this.SmoothedPoints != null)' block and stepped through it. 

The following 'CanonicalSplineHelper.CreateSpline(this.ActualPoints, 0.5, null, false, tolerance)' method receives the correct new list of 'actual' points but returns an empty list of 'smooth' points.

**Comment by RyanBuzzInteractive** [comment]
Also, having followed the call to 'GetNearestInterpolatedPoint' down through DataPointSeries through to XYAxisSeries, I still can't see where the result is actually used anywhere.

**Comment by tibel** [comment]
This issue should be fixed by #499

**Comment by forum111** [comment]
This bug is related with this 
https://github.com/oxyplot/oxyplot/issues/931
